rust/load/src/lib.rs

```diff
@@ -264,44 +264,88 @@ impl PartialEq for Skew {
     }
 }
 
-/////////////////////////////////////////// MetadataQuery //////////////////////////////////////////
+///////////////////////////////////////// TinyStoriesMixin /////////////////////////////////////////
 
-/// A metadata query specifies a metadata filter in Chroma.
-#[derive(Clone, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
-pub enum MetadataQuery {
-    /// A raw metadata query simply copies the provided filter spec.
-    #[serde(rename = "raw")]
-    Raw(serde_json::Value),
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+pub enum TinyStoriesMixin {
+    #[serde(rename = "numeric")]
+    Numeric { ratio_selected: f64 },
 }
 
-impl MetadataQuery {
-    /// Convert the metadata query into a JSON value suitable for use in a Chroma query.
-    pub fn into_where_metadata(self, _: &mut Guacamole) -> serde_json::Value {
+impl TinyStoriesMixin {
+    pub fn to_json(&self, guac: &mut Guacamole) -> serde_json::Value {
         match self {
-            MetadataQuery::Raw(json) => json,
+            Self::Numeric { ratio_selected } => {
+                let field: &'static str = match uniform(0u8, 5u8)(guac) {
+                    0 => "i1",
+                    1 => "i2",
+                    2 => "i3",
+                    3 => "f1",
+                    4 => "f2",
+                    5 => "f3",
+                    _ => unreachable!(),
+                };
+                let mut center = uniform(0, 1_000_000)(guac);
+                let window = (1e6 * ratio_selected) as usize;
+                if window / 2 > center {
+                    center = window / 2
+                }
```

Comment:
I'm wondering if the values for the metadata keys are also generated uniformly at random in range `0..1e6`?

Reply:
I'm assuming so.  Asking Evan after learning time.